### PR TITLE
Add juliaup integration

### PR DESCRIPTION
**Pull request by davidanthoff**
Needs https://github.com/JuliaLang/juliaup/pull/126. This will allow us to bypass this entire messy Julia finding code :)

### CODE REVIEW

**Comment by davidanthoff** [comment]
The `juliaup` prereq shipped now, so this is ready for review and rollout. At the moment `juliaup` is broadly available on Windows and Archlinux, but we're working on all other platforms as well. We could for now disable this on other platforms, but given that there are already (somewhat complicated, admittedly) ways to run `juliaup` on all other platforms as well, I think it makes sense to just ship for all platforms at this point.

File: package.json [hunk 1]
```diff
@@ -729,6 +729,11 @@
                     "description": "Points to the julia executable.",
                     "scope": "machine-overridable"
                 },
+                "julia.juliaupCommand": {
+                    "type": "string",
+                    "default": "juliaup",
+                    "scope": "machine"
```
Comment (pfitzseb) [inline package.json:735]: Can you add a description here? In particular, does this setting allow for version specifiers like `julia +1.6.4`?

Comment (davidanthoff) [reply package.json]: No, this was the path to the `juliaup.exe` binary. I actually removed it now, thinking more about it that is actually not something that should be configurable.

To use a different Julia, it should (in theory) be enough to just put `julia +1.6.4` into the standard exe path setting.

Comment (pfitzseb) [reply package.json]: Ah, makes sense.

**Comment by pfitzseb** [comment]
Needs a rebase, I think.

**Comment by davidanthoff** [comment]
@pfitzseb I think this is ready, if you approve I can merge?

**Review by pfitzseb — APPROVED** [review]

**Comment by rapus95** [comment]
What's the current state of this; Open or merged? 🤔😁

**Comment by davidanthoff** [comment]
And also shipping.

**Pull request merged**